### ISSUE TITLE
Feature request: Add Binary TextClassificationHead ?

### ISSUE THREAD

**Issue by PhilipMay** [deepset-ai/FARM#514]
I am doing binary text classification. To do that I use `TextClassificationHead`. This head has 2 outputs with a softmax function. It works and everything is fine. 

But there is an alternative. Having only one output with a sigmoid function. This is the "normal" way to do binary classification AFAIK.

My own tests with Keras and Image classification showed that this (only one output with sigmoid) proivides better results. But I did not do any extensive tests.

What do you think about this? 
@Timoeller @stefan-it @johann-petrak

**Comment by Timoeller** [comment]
I do not think it will make much of a difference except for stability and speed, which are implementation dependent.

Since CrossEntropyLoss in pytorch combines softmax and crossentropy in a stable way I doubt that there will be much of a difference between the two methods you described. Glad to see experiments proving me wrong. Should not be too hard to code but might require multiple restarts and/or different datasets to compare both methods properly.

**Comment by johann-petrak** [comment]
I *think* it should not really matter, in one case the final layer plus nonlinearity corresponds matematically to  logistic multiclass regression with softmax and in the other to  logistic binary regression with a sigmoid. 

I would be interested in experiments showing a difference that is not due to different random initializations or the like and why that would then actually happen.

**Comment by Timoeller** [comment]
Any decision on if you want to proceed with some tests @PhilipMay ?

**Comment by PhilipMay** [comment]
I think I will not do tests for now. If I do I will reopen this. Closing it...
